tests/unit_tests/pyanaconda_tests/modules/subscription/test_satellite.py

```diff
@@ -0,0 +1,233 @@
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+# Red Hat Author(s): Martin Kolman <mkolman@redhat.com>
+
+import unittest
+from unittest.mock import patch
+from requests import RequestException
+
+from pyanaconda.modules.subscription.satellite import download_satellite_provisioning_script, \
+    run_satellite_provisioning_script, PROVISIONING_SCRIPT_SUB_PATH
+from pyanaconda.core.constants import USER_AGENT, NETWORK_CONNECTION_TIMEOUT
+
+
+class SatelliteLibraryTestCase(unittest.TestCase):
+    """Test the Satellite provisioning code."""
+    # this code at the moment basically just downloads the Satellite provisioning
+    # script from the Satellite instance with one method, then runs it with the other
+
+    @patch("pyanaconda.core.util.requests_session")
+    def script_download_no_prefix_test(self, get_session):
+        """Test the download_satellite_provisioning_script function - no prefix."""
+        # mock the Python Request session
+        session = get_session.return_value.__enter__.return_value
+        result = session.get.return_value
+        result.ok = True
+        result.text = "foo script text"
+        # run the download method
+        script_text = download_satellite_provisioning_script("satellite.example.com")
+        # check script text was returned
+        self.assertEqual("foo script text", script_text)
+        # check the session was called correctly
+        session.get.assert_called_once_with(
+            'http://satellite.example.com' + PROVISIONING_SCRIPT_SUB_PATH,
+            headers={"user-agent": USER_AGENT},
+            proxies={},
+            verify=False,
+            timeout=NETWORK_CONNECTION_TIMEOUT
+        )
+        result.close.assert_called_once()
+
+    @patch("pyanaconda.core.util.requests_session")
+    def script_download_http_test(self, get_session):
+        """Test the download_satellite_provisioning_script function - http prefix."""
+        # mock the Python Request session
+        session = get_session.return_value.__enter__.return_value
+        result = session.get.return_value
+        result.ok = True
+        result.text = "foo script text"
+        # run the download method
+        script_text = download_satellite_provisioning_script("http://satellite.example.com")
```

Comment:
Btw. you can create an additional test that will download a local file using `file://` without mocking the `requests` library. It is useful to confirm that we are using the library correctly.